python/ray/util/sgd/torch/training_operator.py

```diff
@@ -119,17 +119,19 @@ def setup(self, config):
     def __init__(self,
                  config,
                  world_rank,
+                 local_rank,
                  device_ids=None,
                  use_gpu=False,
                  use_fp16=False,
                  use_tqdm=False,
                  apex_args=None,
                  wrap_ddp=False,
-                 wrap_distributed_sampler=False,
+                 use_ddp=False,
                  add_dist_sampler=False,
```

Comment:
what's the difference between these 3 functions?

Reply:
`add_dist_sampler` and `wrap_ddp` come from TorchTrainer and are user provided args. `use_ddp` is internal only and is true if we are using distributed torch runners, and false otherwise. Though I think use_ddp can safely be removed.

Reply:
we actually need use_ddp for the PTL code. But I renamed it to is_distributed to make it clearer.

So just to recap:
add_dist_sampler: User provided arg to determine whether to automatically add Distributed Sampler
wrap_ddp: User provided arg to determine whether to automatically wrap model in DDP
is_distributed: Internal use only. True if Training Operator should do distributed training, False if not.